### PR TITLE
oep(cstorpool) : propose enhancements in cstor pool related APIs.

### PR DESCRIPTION
**Pull request by sonasingh46**
This PR is taken from following 
https://github.com/openebs/openebs/pull/2897

This PR proposes cstorpool related API enhancements.

OpenEBS related API will reside in this repo and hence it made sense to propose the PR here. 
Additionally,  discussions can be done on PR #2897 and any changes that might happen to the API shall be propagated here.
Signed-off-by: Ashutosh Kumar <ashutosh.kumar@mayadata.io>

### CODE REVIEW

File: design/cstor/v1/cstorpool.md [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+# Goals
+
+The major goal of this document is to freeze the schema for CSPC and CSPI schema.
+Apart from this the document focuses on following aspects too:
+
+- Moving the CSPC and CSPI CRD to v1 api version and cstor api group.
+- Enhance the API in order to enhance user experience and support new features.
+
+
+## Current State of Things
```
Comment (kmova) [inline design/cstor/v1/cstorpool.md]: This whole section can be removed. No need for previous version related limitations in this PR.

File: design/cstor/v1/cstorpool.md [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+## Introduction
+
+This proposal highlights the limitations of current CSPC and CSPI schema and
+proposes improvements for the same.
+Please refer to the following link for cstor operator design document. 
+https://github.com/openebs/openebs/blob/master/contribute/design/1.x/cstor-operator/doc.md
+
+# Goals
+
+The major goal of this document is to freeze the schema for CSPC and CSPI schema.
```
Comment (kmova) [inline design/cstor/v1/cstorpool.md]: Here is suggested content for the Goals section:

OpenEBS cStor Data Engine has been in production for over a year now, with multiple cStor custom resource schema revisions. The goal of this design is to incorporate all the feedback received on the alpha and beta versions of cStor custom resources and define a version 1 schema.  

The schema proposed in this document encompasses the following requirements:
- Include the best practicies followed by users with SPC into the schema itself. 
- Rename the SPC and CSP to CSPC and CSPI based on the feedback received. 
- Enhance the elements of CSPC and CSPI to enable performing Day 2 operations declaratively: 
  * Scaling up/down of cStor Pools 
  * Expanding the capacity of cStor Pools 
  * Replacing the failed block devices in cStor Pools. 
  * Add supportability features via Status Conditions and Events 
  * Capture the history / log of upgrades

File: design/cstor/v1/cstorpool.md [hunk 3]
```diff
@@ -0,0 +1,504 @@
+# Moving CSPC and CSPI Schema to cstor group and version v1 from openebs group and version v1alpha1
```
Comment (kmova) [inline design/cstor/v1/cstorpool.md]: Change the title to: Proposed v1 schema for CSPC and CSPI.

File: design/cstor/v1/cstorpool.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    - Scaling up/down of cStor Pools
+    - Expanding the capacity of cStor Pools
+    - Replacing the failed block devices in cStor Pools.
+    - Add supportability features via Status Conditions and Events
+    - Capture the history / log of upgrades
+
+
+## CSPC and CSPI Schema Proposal
+
+###CSPC Schema
```
Comment (kmova) [inline design/cstor/v1/cstorpool.md]: n_ : space after `###`

File: design/cstor/v1/cstorpool.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+type CStorPoolClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []CStorPoolCluster `json:"items"`
+}
+
+```
+
+###CSPI Schema
```
Comment (kmova) [inline design/cstor/v1/cstorpool.md]: n_ : space after `###`

File: design/cstor/v1/cstorpool.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
+# Proposed v1 schema for CSPC and CSPI
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Goals](#goals)
+* [CSPC and CSPI Schema Proposal](#cspc-schema-proposal)
```
Comment (kmova) [inline design/cstor/v1/cstorpool.md]: n_: link is not working. `#cspc-and-cspi-schema-proposal`

Comment (sonasingh46) [reply design/cstor/v1/cstorpool.md]: There was a problem for 3rd bullet. Fixed it.

**Review by kmova — APPROVED** [review]

**Pull request merged**